File: 백준/Silver/16395. 파스칼의 삼각형/파스칼의 삼각형.java
import java.io.*;

public class Main {
    public static void main(String[] args) throws IOException {
        BufferedReader br = new BufferedReader(new InputStreamReader(System.in));
        BufferedWriter bw = new BufferedWriter(new OutputStreamWriter(System.out));

        int result = 0;
        String[] input = br.readLine().trim().split(" ");
        int n = Integer.parseInt(input[0]);
        int k = Integer.parseInt(input[1]);

        int[][] pascal = new int[30][30];
        for (int i = 0; i < 30; i++) {
            pascal[i][0] = 1;
            pascal[i][i] = 1;
        }

        for (int i = 2; i < 29; i++) {
            for (int j = 1; j < i; j++) {
                pascal[i][j] = pascal[i - 1][j - 1] + pascal[i - 1][j];
            }
        }

        result = pascal[n-1][k-1];
        
        bw.write(String.valueOf(result));
        bw.flush();
        br.close();
        bw.close();
    }
}

File: 백준/Silver/16395. 파스칼의 삼각형/파스칼의 삼각형.py
import sys

input = sys.stdin.readline

n, k = map(int, input().strip().split())
pascal = [[0] * 30 for _ in range(30)]

for i in range(30):
    pascal[i][0], pascal[i][i] = 1, 1

for i in range(2, 29):
    for j in range(1, i):
        pascal[i][j] = pascal[i-1][j-1] + pascal[i-1][j]

result = pascal[n-1][k-1]

sys.stdout.write(str(result) + '\n')